### DISCUSSION TITLE
Cloudflare Checkbox

### DISCUSSION THREAD

**Discussion by krypterro** [ultrafunkamsterdam/undetected-chromedriver#1512] [General]
We run a production app with Undetected Chromedriver that logs in to the target sites and reads and writes data, and it does this perfectly 20,000 times an hour for us. Usually we run a 99% success rate, using roughly 20000 Residential Proxies. No problems. 

But on some proxies we get the Cloudflare "click here if you're human" checkbox. It's now actually a widget in a hidden iframe, but pretty easy to isolate and click. Again, no problem, but...

...2 day ago when they switched from just the widget to the iframe, about 5 to 7% of our runs over certain proxies, trigger the checkbox, but when we click it, it says verifying, and then brings us right back to the click checkbox screen. And it's clearly not detecting mouse movement or any other tricky business because even when we manually click it, we get the same results.

Has anyone else had this issue and figured out a way to drive Selenium around it?

**Comment by krypterro** [comment]
The problem is Selenium? I'm not 100% sure what changed exactly, but I know our code to find and click the widget, quit working. When I applied the fix I had use Selenium to switch to the iframe, and click the widget there, which works, but it just refreshs to a new checkbox once it fails verification, and it fails even with the GUI versus headless.

**Comment by krypterro** [comment]
The problem is Selenium? I'm not 100% sure what changed exactly, but I know our code to find and click the widget, quit working. When I applied the fix I had use Selenium to switch to the iframe, and click the widget there, which works, but it just refreshs to a new checkbox once it fails verification, and it fails even with the GUI versus headless.

**Comment by krypterro** [comment]
We are doing it tens of thousands of times per hour, we just get the checkbox when using some proxies. We simply time out the attempts, flag the proxy as bad, and switch the proxy out. But if we could beat the check box again we wouldn't have to burn so many proxies.

**Comment by devCanario** [comment]
this is alredy solved here 
https://github.com/ultrafunkamsterdam/undetected-chromedriver/issues/1543#issuecomment-1709460487

**Comment by devCanario** [comment]
there is a new checkbox again

**Reply by Kamran-ov** [reply]
Hi @canarioPorCanarias,  I saw your solution #1543, however, didn't understand how should I implement document.querySelector(".mark").click(); in python ?  can you please elaborate on your solution?

**Reply by devCanario** [reply]
you need to do 

import undetected_chromedriver as uc
from selenium.webdriver.common.by import By
from time import sleep

driver = uc.Chrome()
driver.get("https://nowsecure.nl/")
sleep(5) # wait for iframe to load 
driver.switch_to.frame(driver.find_element(By.TAG_NAME,"iframe")) # switch to cloudflare iframe
driver.execute_script("document.querySelector('.mark').click();")

**Comment by ShaneOss** [comment]
Doesn't work with https://labs.perplexity.ai :(

**Reply by devCanario** [reply]
you need to use chrome 102

**Reply by n3ling49** [reply]
> you need to use chrome 102

Using the latest chromedriver and version 102.0.5005.61 gets me stuck in a verifying loop for the captcha. Any suggestions?

**Reply by devCanario** [reply]
try using a different  user agent, with maybe last chrome version
